src/libraries/System.Net.Http/tests/FunctionalTests/ResponseStreamZeroByteReadTests.cs

```diff
@@ -0,0 +1,315 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Tests;
+using System.Linq;
+using System.Net.Quic;
+using System.Net.Quic.Implementations;
+using System.Net.Security;
+using System.Net.Test.Common;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.Http.Functional.Tests
```

Comment:
Is there any way we could just use stream conformance tests directly here?

We have some support for running stream conformance tests against response streams already. Can we just improve that so it (a) enables zero byte read tests and (b) covers all the cases we want it to?

Reply:
Technically, yes, but it would require changes in the base conformance test class and our loopback servers.

The main issues I hit here were:
- The connected stream tests don't advertise the amount of data that will be transferred to different implementations. For `Content-Length`, we have to know the number upfront.
- The generic loopback server does not support partial response content read/writes, without which we can't have a stream wrapping a `GenericLoopbackConnection`.

We can certainly address all of such problems and turn all of these into `StreamConformanceTests`, I just wasn't convinced it was worth doing as part of this PR.